src/realm/object-store/object_schema.hpp

```diff
@@ -62,7 +66,9 @@ class ObjectSchema {
     std::vector<Property> computed_properties;
     std::string primary_key;
     TableKey table_key;
+    // Cannot be both true at the same time.
```

Comment:
we can only set at most one through our constructors, so it's more for future reference.

Reply:
if these are meant to only be settable via constructors, should we make them as const?

Reply:
I just realized all members are public and the JS SDK sets them directly. Making them const will break JS as it is now, although it shouldn't be a big change to start using the constructors (as it should). An alternative, as @tgoyne suggested, is to turn the member into an enum, but I feel this might have some implications when binding with other SDKs.